pandas/tests/groupby/test_quantile.py

```diff
@@ -248,6 +248,33 @@ def test_groupby_quantile_skips_invalid_dtype(q):
     tm.assert_frame_equal(result, expected)
 
 
+def test_groupby_quantile_NA_float(any_float_dtype):
+    # GH#42849
+    df = DataFrame({"x": [1, 1], "y": [0.2, np.nan]}, dtype=any_float_dtype)
+    result = df.groupby("x")["y"].quantile(0.5)
+    expected = pd.Series([0.2], dtype=float, index=[1.0], name="y")
+    expected.index.name = "x"
+    tm.assert_series_equal(expected, result)
+
+
+def test_groupby_quantile_NA_int(any_int_ea_dtype):
+    # GH#42849
+    df = DataFrame({"x": [1, 1], "y": [2, 5]}, dtype=any_int_ea_dtype)
+    result = df.groupby("x")["y"].quantile(0.5)
+    expected = pd.Series([3.5], dtype=float, index=[1], name="y")
+    expected.index.name = "x"
+    tm.assert_series_equal(expected, result)
+
+
+def test_groupby_quantile_allNA_column():
+    # GH#42849
+    df = DataFrame({"x": [1, 1], "y": [pd.NA] * 2}, dtype="Float64")
```

Comment:
any reason for this to be Float64 instead of any_foo_dtype?

Reply:
not sure what `any_foo_dtype` means. 
In case it means why the explicit dtype? then without explicitly defining with nullable dtypes, the columns becomes `object`, and quantile fails (maybe I am wrong, but I believe this is the expected behaviour). Below on master.

```python
In [3]: import pandas as pd

In [4]: pd.__version__
Out[4]: '1.4.0.dev0+540.ga826be1f61'

In [5]: DataFrame({"x": [1, 1], "y": [pd.NA] * 2}).dtypes
Out[5]: 
x     int64
y    object
dtype: object

In [6]: DataFrame({"x": [1, 1], "y": [pd.NA] * 2}, dtype=float).dtypes
<ipython-input-6-4731f064b6c6>:1: FutureWarning: Could not cast to float64, falling back to object. This behavior is deprecated. In a future version, when a dtype is passed to 'DataFrame', either all columns will be cast to that dtype, or a TypeError will be raised.
  DataFrame({"x": [1, 1], "y": [pd.NA] * 2}, dtype=float).dtypes
Out[6]: 
x    float64
y     object
dtype: object